src/pipeline.cpp

```diff
@@ -125,7 +126,7 @@ Status Pipeline::execute() {
             IF_ERROR_OCCURRED_EARLIER_THEN_BREAK_IF_ALL_STARTED_FINISHED_CONTINUE_OTHERWISE
             SessionResults sessionResults;
             SPDLOG_LOGGER_DEBUG(dag_executor_logger, "Fetching results of pipeline: {} node: {} session: {}", getName(), finishedNode.getName(), sessionKey);
-            status = finishedNode.fetchResults(sessionKey, sessionResults);
+            status = finishedNode.fetchResults(sessionKey, sessionResults, context);
```

Comment:
I think you should pass context in L92 Into NodeSessionMetadata. Then you avoid passing it through all interfaces of DAG's.

Reply:
Moved to metadata